pkg/services/alerting/notifiers/alertmanager.go

```diff
@@ -182,10 +184,16 @@ func (am *AlertmanagerNotifier) Notify(evalContext *alerting.EvalContext) error
 
 		if err := bus.DispatchCtx(evalContext.Ctx, cmd); err != nil {
 			am.log.Error("Failed to send alertmanager", "error", err, "alertmanager", am.Name, "url", url)
-			return err
+			errCnt++
 		}
 	}
 
+	// This happens when every dispatch return error
+	if errCnt == len(am.URL) {
+		am.log.Error("Failed to send alert for all urls")
+		return fmt.Errorf("failed to send alert for all urls")
```

Comment:
Think we can skip logging the error in here and just return it

```suggestion
		return fmt.Errorf("failed to send alert to alertmanager")
```

Reply:
Thanks for your suggestion! Should be resolved :)